detectable/src/main/java/com/synopsys/integration/detectable/detectables/yarn/parse/YarnTransformer.java

```diff
@@ -45,35 +42,26 @@ public YarnTransformer(final ExternalIdFactory externalIdFactory) {
         this.externalIdFactory = externalIdFactory;
     }
 
-    public DependencyGraph transform(List<YarnListNode> yarnList, YarnLock yarnLock){
-        final MutableDependencyGraph graph = new MutableMapDependencyGraph();
-        final DependencyHistory history = new DependencyHistory();
+    public DependencyGraph transform(final PackageJson packageJson, final YarnLock yarnLock) {
+        final LazyExternalIdDependencyGraphBuilder graphBuilder = new LazyExternalIdDependencyGraphBuilder();
 
-        for (YarnListNode yarnListNode : yarnList){
-            int actualDepth = yarnListNode.getDepth() - 1;//"yarn list" is the root node, so we need to shift the entire tree 1 level to the left.
-            try {
-                history.clearDependenciesDeeperThan(actualDepth);
-            } catch (final IllegalStateException e) {
-                logger.warn(String.format("Problem parsing yarn list '%s': %s", yarnListNode.getFuzzyId(), e.getMessage()));
-            }
-
-            String name = yarnListNode.getPackageName();
-            Optional<String> resolvedVersion = yarnLock.versionForFuzzyId(yarnListNode.getFuzzyId());
-            String version = resolvedVersion.orElse(yarnListNode.getFuzzyPackageVersion());
-
-            final ExternalId externalId = externalIdFactory.createNameVersionExternalId(Forge.NPMJS, name, version);
-            Dependency dependency = new Dependency(name, version, externalId);
+        for (final Map.Entry<String, String> packageDependency : packageJson.dependencies.entrySet()) {
+            graphBuilder.addChildToRoot(new StringDependencyId(packageDependency.getKey() + "@" + packageDependency.getValue()));
+        }
 
+        for (final Map.Entry<String, String> packageDependency : packageJson.devDependencies.entrySet()) {
+            graphBuilder.addChildToRoot(new StringDependencyId(packageDependency.getKey() + "@" + packageDependency.getValue()));
+        }
```

Comment:
It looks like we are always adding devDependencies. Since yarn has a production only flag, I don't think we should be doing this.